rfcs/0193-toml-flakes.md

```diff
@@ -0,0 +1,253 @@
+---
+feature: toml-flakes
+start-date: 2025-12-07
+author: Robert Hensing (@roberth)
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Make flakes easier to use, automate and learn.
+
+Introduce `flake.toml` as the new leading file for flakes,
+separating input specifications from output definitions.
+The `flake.toml` file contains declarative metadata and input declarations,
+while `flake.nix` focuses on output definitions.
+
+Fixes [#4945 What language is flake.nix written in?](https://github.com/NixOS/nix/issues/4945)
+
+# Motivation
+[motivation]: #motivation
+
+Currently, flakes combine two distinct concerns in a single `flake.nix` file:
+declaring dependencies (inputs) and implementing functionality (outputs).
+This creates several problems:
+
+1. **User uncertainty**: The current structure creates confusion about what language features are available
+   and when evaluation restrictions apply,
+   because inputs and outputs are mixed in the same file, and inputs are not allowed to be arbitrary Nix expressions.
```

Comment:
Adding a second language for flakes doesn't make it easier for new users to understand flakes, since they now need to learn how to use flakes in two languages instead of one.